src/Symfony/Component/Security/Http/RateLimiter/DefaultLoginRateLimiter.php

```diff
@@ -28,21 +28,36 @@ final class DefaultLoginRateLimiter extends AbstractRequestRateLimiter
 {
     private RateLimiterFactory $globalFactory;
     private RateLimiterFactory $localFactory;
+    private ?string $secret;
 
-    public function __construct(RateLimiterFactory $globalFactory, RateLimiterFactory $localFactory)
+    /**
+     * @param non-empty-string|null $secret A secret to use for hashing the IP address and username
```

Comment:
no need to document the deprecated way:
```suggestion
     * @param non-empty-string $secret A secret to use for hashing the IP address and username
```